test/xbgpu/test_bsend.py

```diff
@@ -89,82 +94,101 @@ async def _send_data(
             chunk = await send_stream.get_free_chunk()
 
+            # Generate random data for the i-th chunk
+            rng = np.random.default_rng(seed=1)
+            data[i, ...] = rng.integers(low=-128, high=127, size=(data.shape[1:]), dtype=np.int8)
+
             # Populate the buffer with dummy data.
-            chunk.data.fill(0)
+            chunk.data[:] = data[i, ...]
 
             # Give the chunk back to the send_stream to transmit out
             # onto the network.
-            chunk.timestamp = i * HEAPS_PER_FENG_PER_CHUNK * heap_timestamp_step
+            chunk.timestamp = i * HEAPS_PER_CHUNK * heap_timestamp_step
             send_stream.send_chunk(chunk, time_converter, sensors)
         # send_heap just queues data for sending but is non-blocking.
         # Flush to ensure that the data all gets sent before we return.
         await send_stream.stream.async_flush()
 
+        return data
+
     @staticmethod
     async def _recv_data(
+        data: np.ndarray,
+        outputs: Sequence[BOutput],
         queues: list[spead2.InprocQueue],
+        n_engines: int,
+        engine_id: int,
         channel_offset: int,
         n_channels_per_substream: int,
         n_spectra_per_heap: int,
         heap_timestamp_step: int,
     ) -> None:
-        """Receive data transmitted from :func:`_send_data`.
+        """Receive data transmitted from :meth:`_send_data`.
 
         Error-check data here as well.
 
         Parameters
         ----------
-        outputs
-            Fixture
+        data
+            Random data generated during data transmission, of shape
+            - (N_TX_ITEMS, HEAPS_PER_CHUNK, len(outputs), n_channels_per_substream, n_spectra_per_heap, COMPLEX)
         queues
             List of :class:`spead2.InprocQueue` used to transmit heaps
             in :meth:`_send_data`.
-        channel_offset, n_channels_per_substream, n_spectra_per_heap, heap_timestamp_step
-            Variables declared by the calling unit test to verify
-            transmitted data.
         """
-        out_config = spead2.recv.StreamConfig(max_heaps=100)
+        # Reshape as we verify *heaps* per substream, not chunks
+        data = data.reshape(
+            (TOTAL_DATA_HEAPS_PER_SUBSTREAM, len(outputs), n_channels_per_substream, n_spectra_per_heap, COMPLEX)
+        )
+
+        out_config = spead2.recv.StreamConfig()
         out_tp = spead2.ThreadPool()
-        for queue in queues:
+        for i, queue in enumerate(queues):
             stream = spead2.recv.asyncio.Stream(out_tp, out_config)
             stream.add_inproc_reader(queue)
 
             # Wait for the first packet to arrive - it is expected to be the
             # SPEAD descriptor.
             ig = spead2.ItemGroup()
             heap = await stream.get()
-            # TODO: No checks on the count sequence yet
+            assert heap.cnt % n_engines == engine_id % n_engines, "The heap IDs are not correctly strided"
```

Comment:
```suggestion
            assert heap.cnt % n_engines == engine_id, "The heap IDs are not correctly strided"
```
Since engine_id should be in the range [0, n_engines).